specification/peering/resource-manager/Microsoft.Peering/preview/2019-09-01-preview/peering.json

```diff
@@ -328,46 +336,521 @@
+          },
+          {
+            "name": "peeringName",
+            "in": "path",
+            "description": "The name of the peering.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "registeredAsnName",
+            "in": "path",
+            "description": "The name of the registered ASN.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a registered ASN associated with the peering": {
+            "$ref": "./examples/GetRegisteredAsn.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PeeringRegisteredAsns"
+        ],
+        "description": "Creates a new registered ASN with the specified name under the given subscription, resource group and peering.",
+        "operationId": "RegisteredAsns_CreateOrUpdate",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The name of the resource group.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "peeringName",
+            "in": "path",
+            "description": "The name of the peering.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "registeredAsnName",
+            "in": "path",
+            "description": "The name of the ASN.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "registeredAsn",
+            "in": "body",
+            "description": "The properties needed to create a registered ASN.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/PeeringRegisteredAsn"
+            }
```

Comment:
is it long running ?

If so, please mark  "x-ms-long-running-operation": true,  
Update this for all 201/201 LROs.

Also, to let SDK know how to track.. add "x-ms-long-running-operation-options" extension with "final-state-via" property.
``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "location"
}
```

or

``` json
"x-ms-long-running-operation-options": {
  "final-state-via": "azure-async-operation"
}
```